code/modules/client/client_procs.dm

```diff
@@ -1128,8 +1128,8 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 		winset(src, "default.Shift", "is-disabled=true")
 		winset(src, "default.ShiftUp", "is-disabled=true")
 
-/client/proc/update_ambience_pref()
-	if(prefs.read_preference(/datum/preference/toggle/sound_ambience))
+/client/proc/update_ambience_pref(value)
+	if(value || prefs.read_preference(/datum/preference/toggle/sound_ambience))
```

Comment:
First of all - wtf `apply_to_client` is called before the mob even gets prefs???

Second of all - if `value` is `FALSE`, won't this still runtime?

Reply:
Maybe we just use `apply_to_client_updated`

Reply:
> First of all - wtf `apply_to_client` is called before the mob even gets prefs???
apply_all_client_preferences() is what's SENDING the value of the preference during initial client/new, later calls can properly use prefs.red_preference
https://github.com/tgstation/tgstation/blob/2f2b74338cb7397e4128ccdef00bbc4d6c67a0a5/code/modules/client/client_procs.dm#L242-L273
https://github.com/tgstation/tgstation/blob/2f2b74338cb7397e4128ccdef00bbc4d6c67a0a5/code/modules/client/preferences.dm#L329-L337
> 
> Second of all - if `value` is `FALSE`, IE they have the pref toggled off, won't this still runtime?
Only the very first call of a player's savefile is the issue here, after that the preference is set.



>Maybe we just use apply_to_client_updated()
![image](https://github.com/tgstation/tgstation/assets/6209658/49593788-5742-4c40-b385-aa479c0d41e5)
same issue.

Reply:
infact one sec, can just eliminate this check completely.